utils/collect-updates/collect-updates.js

```diff
@@ -65,21 +65,5 @@ function collectUpdates({
   const dependents = collectDependents(candidates);
   dependents.forEach(node => candidates.add(node));
 
-  if (canary || packages.size === candidates.size) {
-    logger.verbose("updated", "(short-circuit)");
-
-    return Array.from(candidates);
-  }
-
-  const updates = [];
-
-  packages.forEach((node, name) => {
-    if (candidates.has(node)) {
-      logger.verbose("updated", name);
-
-      updates.push(node);
-    }
-  });
-
-  return updates;
+  return Array.from(candidates);
```

Comment:
Indeed, simply remove the short-circuit entirely and no tests need change at all:
```patch
diff --git a/utils/collect-updates/collect-updates.js b/utils/collect-updates/collect-updates.js
index 300655f2..e90f2191 100644
--- a/utils/collect-updates/collect-updates.js
+++ b/utils/collect-updates/collect-updates.js
@@ -65,12 +65,7 @@ function collectUpdates({
   const dependents = collectDependents(candidates);
   dependents.forEach(node => candidates.add(node));
 
-  if (canary || packages.size === candidates.size) {
-    logger.verbose("updated", "(short-circuit)");
-
-    return Array.from(candidates);
-  }
-
+  // The result should always be in the same order as the input
   const updates = [];
 
   packages.forEach((node, name) => {
```